src/dswx_sar/filter_SAR.py

```diff
@@ -122,11 +125,210 @@ def lee_enhanced_filter(img, win_size=3, k=K_DEFAULT, cu=CU_DEFAULT,
     filter_im: numpy.ndarray
         filtered intensity image.
     """
-    print('>> lee_enhanced_filter')
+    print('>> lee_enhanced_filter', kwargs)
+    win_size = kwargs.get('win_size', -1)
 
     # we process the entire img as float64 to avoid type overflow error
     img = np.float64(img)
     w_t, mean, _ = weightingarr(img, win_size, k, cu, cmax)
     filter_im = (mean * w_t) + (img * (1 - w_t))
 
     return filter_im
+
+
+def fill_nan_value(data):
+    mask = np.isnan(data) | np.isinf(data) | np.isinf(-data)
+    data[mask] = -15
+    return data
+
+
+def find_largest_nan_polygon(data):
+    """
+    Identifies contiguous NaN regions and finds the largest one.
+
+    :param data: 2D numpy array with NaNs.
+    :return: The label of the largest NaN region and an array where each
+             contiguous NaN region is labeled with a unique integer.
+    """
+    # Identify NaNs and invert, because ndimage.label labels non-zero regions
+    nan_mask = np.isnan(data)
+    labeled_regions, num_features = ndimage.label(~nan_mask)
+
+    # Count the occurrences of each label (excluding the background, which is 0)
+    label_sizes = np.bincount(labeled_regions.ravel())[1:]  # Exclude label 0
+
+    # Find the label of the largest NaN region
+    largest_label = np.argmax(label_sizes) + 1  # +1 because we excluded label 0 earlier
+
+    return largest_label, labeled_regions
+
+
+def interpolate_except_largest_nan_region(data):
```

Comment:
A question: Is the "largest NaN area" going to be the void area surrounding the image at the center?

Reply:
@seongsujeong Sorry for confusion. I'll delete this function. It was designed to avoid the Nan value issues in filtering, but we don't need it for now.